test/views/declaration/DeclarantDetailsViewSpec.scala

```diff
@@ -56,61 +56,72 @@ class DeclarantDetailsViewSpec extends UnitViewSpec with ExportsTestData with Co
     onEveryDeclarationJourney() { implicit request =>
       "display page title" in {
 
-        createView(form(request.declarationType))
+        createView(form())
           .getElementsByClass(Styles.gdsPageLegend) must containMessageForElements("declaration.declarant.titleQuestion", request.eori)
       }
 
-      "display section header" in {
-
-        createView(form(request.declarationType)).getElementById("section-header") must containMessage("declaration.section.2")
-      }
-
       "display radio button with Yes option" in {
 
-        val view = createView(form(request.declarationType))
+        val view = createView(form())
         view.getElementById("code_yes").attr("value") mustBe YesNoAnswers.yes
         view.getElementsByAttributeValue("for", "code_yes") must containMessageForElements("site.yes")
       }
       "display radio button with No option" in {
 
-        val view = createView(form(request.declarationType))
+        val view = createView(form())
         view.getElementById("code_no").attr("value") mustBe YesNoAnswers.no
         view.getElementsByAttributeValue("for", "code_no") must containMessageForElements("site.no")
       }
 
       "display 'Save and continue' button on page" in {
 
-        val saveButton = createView(form(request.declarationType)).getElementById("submit")
+        val saveButton = createView(form()).getElementById("submit")
         saveButton must containMessage(saveAndContinueCaption)
       }
+    }
 
-      "display 'Save and return' button on page" in {
+    onJourney(STANDARD, SUPPLEMENTARY, SIMPLIFIED, OCCASIONAL) { implicit request =>
+      "display section header" in {
 
-        val saveButton = createView(form(request.declarationType)).getElementById("submit_and_return")
-        saveButton must containMessage(saveAndReturnCaption)
-        saveButton.attr("name") mustBe SaveAndReturn.toString
+        createView(form()).getElementById("section-header") must containMessage("declaration.section.1")
       }
-    }
 
-    onJourney(STANDARD, SUPPLEMENTARY, SIMPLIFIED, OCCASIONAL) { implicit request =>
-      "display 'Back' button that links to 'Consignment References' page" in {
+      "display 'Back' button that links to 'Additional Declaration Type' page" in {
 
-        val view = declarantDetailsPage(Mode.Normal, form(request.declarationType))(request, messages)
+        val view = declarantDetailsPage(Mode.Normal, form())(request, messages)
         val backButton = view.getElementById("back-link")
 
         backButton must containMessage(backCaption)
-        backButton.attr("href") mustBe routes.ConsignmentReferencesController.displayPage().url
+        backButton must haveHref(routes.AdditionalDeclarationTypeController.displayPage().url)
+      }
+
+      "not display 'Save and return' button on page" in {
+
+        val saveButton = createView(form()).getElementById("submit_and_return")
+        saveButton mustBe null
```

Comment:
Why not `Option(createView(form()).getElementById("submit_and_return")) mustBe None` ?

Reply:
I saw in some other place we did it this way. Besides it is a bit more indirect testing - `getElementById` returns `null` if element is not found (as it has Java origins). Not sure about wrapping it in `Option` just to avoid typing `null`. If it was application code then surely this is something to avoid, but in tests I think it is fine.
I'm leaving it as it is.

Reply:
Or `Option(saveButton) mustBe None`

Reply:
:+1: